wpilib/wpilib/cantalon.py

```diff
@@ -168,6 +168,7 @@ def handle(self):
         return self._handle
 
     def free(self):
+        LiveWindow.removeComponent(self)
```

Comment:
I don't think so, because LiveWindow components are stored in a dictionary with the device objects as the keys of the dictionary. You need to have the device object in order to remove the entry from that dictionary, and the finalizer doesn't have access to that object, since it is supposed to run after the object has been deleted.